### PR TITLE
SF-3023 Add System Admin Authorization Filter for Hangfire

### PR DESCRIPTION
**Pull request by pmachapman**
This PR changes the Hangfire Dashboard so that it can only be accessed by a logged in System Administrator, and adds a Hangfire Dashboard menu item to the top right User menu.

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2773)

### CODE REVIEW

**Review by kylebuss — COMMENTED** [review]
This is great! Just curious what the reasoning is for the 30 minute expiry token?

> Reviewed 8 of 9 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2773#-:-O8bKUin9gEyn61OcR5S:b-m4hphe)* status: 8 of 9 files reviewed, 1 unresolved discussion (waiting on @pmachapman)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/app.component.ts` line 378 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2773#-O8bHmHW2QBxmvmFxU3L:-O8bHmHW2QBxmvmFxU3M:b28d4u6) ([raw file](https://github.com/sillsdev/web-xforge/blob/6398d63192f874f0130f988d324fec13c4176cda/src/SIL.XForge.Scripture/ClientApp/src/app/app.component.ts#L378)):*
> ```XML
>       expires.setMinutes(expires.getMinutes() + 30);
>       this.cookieService.set('Hangfire', accessToken, expires, '/hangfire');
>       this.locationService.go('/hangfire');
> ```

nit I find myself typically opening Hangfire in a seperate tab. I wonder if we should do that by default?

Adding a function to LocationService could look like:

```typescript
openInNewTab(url: string) {
   window.open(url, '_blank');
}
```



_Code quote:_
```XML
this.locationService.go('/hangfire');
```

**Review by pmachapman — COMMENTED** [review]
> This is great! Just curious what the reasoning is for the 30 minute expiry token?

Mostly because I had to choose a time, and 30 minutes seemed long enough (I don't want the cookie hanging around too long). Happy to change it if you like.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2773#-:-O8cGhFe-ldzjdTzGwO3:b-7y5n8p)* status: 7 of 10 files reviewed, all discussions resolved (waiting on @kylebuss)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/app.component.ts` line 378 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2773#-O8bHmHW2QBxmvmFxU3L:-O8cJyjA7eOKFOopB2QC:bb3jycm) ([raw file](https://github.com/sillsdev/web-xforge/blob/6398d63192f874f0130f988d324fec13c4176cda/src/SIL.XForge.Scripture/ClientApp/src/app/app.component.ts#L378)):*
<details><summary><i>Previously, kylebuss (Kyle Buss) wrote…</i></summary><blockquote>

nit I find myself typically opening Hangfire in a seperate tab. I wonder if we should do that by default?

Adding a function to LocationService could look like:

```typescript
openInNewTab(url: string) {
   window.open(url, '_blank');
}
```
</blockquote></details>

Done. Thanks for the idea!

**Review by kylebuss — APPROVED** [review]
Gotcha. 30 minutes should be long enough to investigate any issues that would show in Hangfire. I came from an app that Admin users could view Hangfire in any environment if you were logged in and typed in the url, but I really like the menu item for it! 

> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2773#-:-O8cP-dc6o7de_ysgOL7:bgzte36)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @pmachapman)

File: src/SIL.XForge.Scripture/ClientApp/src/app/app.component.html [hunk 1]
```diff
@@ -132,6 +132,12 @@
               {{ t("serval_administration") }}
             </button>
           }
+          @if (isSystemAdmin) {
+            <button mat-menu-item id="hangfire-dashboard-btn" [disabled]="!isAppOnline" (click)="hangfireDashboard()">
+              <mat-icon>pending_actions</mat-icon>
+              {{ t("hangfire_dashboard") }}
```
Comment (Nateowami) [inline src/SIL.XForge.Scripture/ClientApp/src/app/app.component.html:138]: I realize you're just following the pattern, but as I was about to upload the latest strings to Crowdin, I'm wondering how translators are supposed to translate this, or why we would ask them to. Same could be said for a couple of the others.

**Pull request merged**